text/1029-deprecate-app-prefix.md

```diff
@@ -0,0 +1,108 @@
+
+
+### Deprecation Guide Content
+
+#### app-prefix
+
+Returning content from an addon's `contentFor()` hook for `type="app-prefix"` is deprecated. Addons will no longer be allowed to inject arbitrary javascript here. If you need to provide code that apps will run before booting, document that app authors should import and call your code at the start of their own `app.js` file.
+
+#### app-suffix
+
+Returning content from an addon's `contentFor()` hook for `type="app-suffix"` is deprecated. Addons will no longer be allowed to inject arbitrary javascript here. If you need to provide code that apps will run before booting, document that app authors should import and call your code at the start of their own `app.js` file.
+
+If you were using app-suffix to overwrites modules provided by the app, that is intentionally not supported. Adjust your API to tell app authors to import your code and invoke it where appropriate.
+
+#### tests-prefix
+
+Returning content from an addon's `contentFor()` hook for `type="tests-prefix"` is deprecated. Addons will no longer be allowed to inject arbitrary javascript here. Provide utilities that users can import into their own test setup code instead.
+
+#### tests-suffix
+
+Returning content from an addon's `contentFor()` hook for `type="tests-suffix"` is deprecated. Addons will no longer be allowed to inject arbitrary javascript here. Provide utilities that users can import into their own test setup code instead.
+
+#### vendor-prefix
+
+Returning content from an addon's `contentFor()` hook for `type="vendor-prefix"` is deprecated. Addons will no longer be allowed to inject arbitrary javascript here. If you really need to run script (non-module) code, provides your own script via your addon's `/public` directory and either document that app authors should createa a `<script>` element in their HTML that includes it, or use `contentFor()` with one of the `type`s that appears in `index.html` to emit the scrip tag automatically. (`contentFor` targeting HTML is not deprecated, this deprecation only covers targeting javascript bundles.)
+
+#### vendor-suffix
+
+Returning content from an addon's `contentFor()` hook for `type="vendor-suffix"` is deprecated. Addons will no longer be allowed to inject arbitrary javascript here. If you really need to run script (non-module) code, provides your own script via your addon's `/public` directory and either document that app authors should createa a `<script>` element in their HTML that includes it, or use `contentFor()` with one of the `type`s that appears in `index.html` to emit the scrip tag automatically. (`contentFor` targeting HTML is not deprecated, this deprecation only covers targeting javascript bundles.)
+
+## Drawbacks
+
+This is a change to the v1 addon API. V2 addons already cannot use contentFor. One could argue against bothering to change the v1 addon API.
```

Comment:
> V2 addons already cannot use contentFor

Is this referring to the specific hooks this RFC aims to deprecate? As according to [the spec in the Embroider repository](https://github.com/embroider-build/embroider/blob/d9fc688f7abd5663045e182da9f28a44c84cfc64/docs/spec.md#contentfor), regular `contentFor` hooks are still supported.

Should this be reworded?

```suggestion
This is a change to the v1 addon API. V2 addons already cannot use the contentFor hooks this RFC aims to deprecate. One could argue against bothering to change the v1 addon API.
```

Reply:
Oh, interesting, didn't know _that_ spec was saying that. But the _actual_ v2 format RFC is not saying that at all, it is saying that `contentFor` is not available: https://rfcs.emberjs.com/id/0507-embroider-v2-package-format/#scope-of-this-rfc. 

Which matches what we have in reality I think. While you _could_ add `contentFor` to the v1 shim (`addon-main.cjs`), you could add any other v1 hook there as well, but this does not make it a supported feature, and does not work under native Embroider AFAIK. So I think this RFC is correct here.

Reply:
That makes sense. I was confused due to the docs in that repo, as I also had in mind `contentFor` was unsupported in v2.

I've opened a ticket to keep track of updating those docs. (https://github.com/embroider-build/embroider/issues/2083)